service/storage_service.cc

```diff
@@ -398,7 +398,11 @@ future<> storage_service::sync_raft_topology_nodes(mutable_token_metadata_ptr tm
         // Also ip -> id mapping is needed for address map recreation on reboot
         if (is_me(host_id)) {
             co_await _sys_ks.local().update_tokens(rs.ring.value().tokens);
-            co_await _gossiper.add_local_application_state({{ gms::application_state::STATUS, gms::versioned_value::normal(rs.ring.value().tokens) }});
+            co_await _gossiper.add_local_application_state({
+                { gms::application_state::TOKENS, gms::versioned_value::tokens(rs.ring.value().tokens) },
+                { gms::application_state::CDC_GENERATION_ID, gms::versioned_value::cdc_generation_id(_topology_state_machine._topology.current_cdc_generation_id) },
+                { gms::application_state::STATUS, gms::versioned_value::normal(rs.ring.value().tokens) }
+            });
```

Comment:
It's a step in the wrong direction. We should have been removing things from gossiper not adding them back. Even having a state here is a hack :(

Reply:
Once we'll implement another kind of recovery (using external tool to calculate new topology instead of going through gossip-based removenode), we'll be able to get rid of this.